### ISSUE TITLE
variables in foreach loop can be erroneously self-assigned to suppress UnusedVariable

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1391]
# issue
whilst `$a = $a;` still allows `UnusedVariable` to bubble up, `$k = $k; $v = $v;` where `foreach ($arr as $k => $v){ .. }```, stops `UnusedVariable` from bubbling up through the error list.

# actual
Assigning a variable from a `foreach` loop to itself does not trigger `UnusedVariable`

# expected
Assigning a variable from a `foreach` loop to itself does trigger `UnusedVariable`

# repro

* https://psalm.dev/r/c79d386c2c
* 4f770d3

```php
<?php
$a = 1;
$a = $a;

$b = [1,2,3];

foreach ($b as $c) {
  $c = $c;
}

foreach ($b as $k => $v) {
  $k = $k;
  $v = $v;
}
```
> INFO: UnusedVariable - 3:1 - Variable $a is never referenced

**Comment by muglug** [comment]
Bug, but a low priority one
